### ISSUE TITLE
Wrong typing for queryMethod

### ISSUE THREAD

**Issue by Apokalypt** [typegoose/typegoose#665]
## Versions

- System:  windows 
- NodeJS: 16.11.0
- Typescript: 4.5.5
- Compiler / Transpiler: tsc 
- Typegoose(NPM): 9.5.0
- mongoose: 6.2.0

## What is the Problem?

The current typing for queryMethod decorator is invalid. When the query method is called, the "this" property isn't a ReturnModelType<any> instance.
Therefore, when we call some methods it is not the right methods that are identified by Typescript, more information : https://discord.com/channels/612400042900193311/612406252164612129/939208028119531550

## Do you know *why* it happens?

Query methods can only be called when mongoose returns a "QueryWithHelpers" and not directly on model due to the implementation.

**Comment by Apokalypt** [comment]
Instead of using 
```
ReturnModelType<typeof Class, Helpers>
```
We should use :
```
type ResultDoc = DocumentType<typeof Class, Helpers>;

QueryWithHelpers<ResultDoc | null, ResultDoc, Helpers, typeof BuildingGameElement>
```

**Comment by hasezoey** [comment]
This was fixed with d1215282eb761aabcfcfcb4f1d66808f1ab04cee, which introduces a new type `QueryHelperThis` for convenience to work like `DocumentType` and `ReturnModelType`
